src/pip/_internal/commands/show.py

```diff
@@ -122,47 +148,96 @@ def search_packages_info(query):
+def get_package_info(dist, list_files, verbose):
+    """
+    Gather details from an installed distribution.
+    """
+    name = dist.get('name', '')
+    required_by = [
+        pkg.project_name for pkg in pkg_resources.working_set
+        if name in [required.name for required in pkg.requires()]
+    ]
+    info = [
+        ("Name", name),
+        ("Version", dist.get('version', '')),
+        ("Summary", dist.get('summary', '')),
+        ("Home-page", dist.get('home-page', '')),
+        ("Author", dist.get('author', '')),
+        ("Author-email", dist.get('author-email', '')),
+        ("License", dist.get('license', '')),
+        ("Location", dist.get('location', '')),
+        ("Requires", dist.get('requires', [])),
+        ("Required-by", required_by)
+    ]
+
+    if verbose:
+        classifiers = [str(classifier)
+                       for classifier in dist.get('classifiers', [])]
+
+        parser = configparser.ConfigParser()
+
+        parser.readfp(StringIO(u'\n'.join(dist.get('entry_points', []))))
+
+        entry_points = {section: dict(parser.items(section))
+                        for section in parser.sections()}
+        info.extend([
+            ("Metadata-Version", dist.get('metadata-version', '')),
+            ("Installer", dist.get('installer', '')),
+            ("Classifiers", classifiers),
+            ("Entry-points", entry_points)
+        ])
+
+    if list_files:
+        if "files" not in dist:
+            info.extend([("Files", None)])
+        else:
+            files = [str(line.strip()) for line in dist.get('files ', [])]
+            info.extend([("Files", files)])
+    return info
+
+
+def get_distributions_info(distributions, list_files=False, verbose=False):
     """
-    Print the informations from installed distributions found.
+    Gather information for all installed distributions.
     """
-    results_printed = False
-    for i, dist in enumerate(distributions):
-        results_printed = True
-        if i > 0:
+    info = []
+    have_distributions_info = False
+
+    for dist in distributions:
+        have_distributions_info = True
+
+        package_info = get_package_info(dist, list_files, verbose)
+
+        info.append(OrderedDict(package_info))
+    return info, have_distributions_info
+
+
+def print_header_format(info):
+    """
+    Print the information from installed distributions found.
+    """
+    for package in info:
+        for key, value in package.items():
+            if key == 'Classifiers':
+                logger.info("%s:", key)
+                for classifier in value:
+                    logger.info("  %s", classifier)
+            elif key == 'Entry-points':
+                logger.info("%s:", key)
+                for entry_point, entry_point_info in value.items():
+                    logger.info("  [%s]", entry_point)
+                    for x, y in entry_point_info.items():
+                        logger.info("  %s = %s", x, y)
```

Comment:
It looks like you don't have the `logger.info("Files:")` line anymore.

Reply:
For JSON output, entry points data is needed in a structured format (i.e. a dict of dicts), whereas for text output the entry points data is just dumped out as is. `ConfigParser` does the TOML parsing of the entry points data.

Reply:
Are you replying to the `logger.info("Files:")` comment or the `ConfigParser` comment? You're referencing `ConfigParser` in your comment but it's showing up in reply to the `logger.info("Files:")` comment.

Reply:
Yes, sorry, no idea why this ended up under the wrong comment.

Reply:
I'll delete my comments here.